framework/aster-frame/src/sync/rcu/mod.rs

```diff
@@ -70,35 +159,49 @@ pub struct RcuReclaimer<P> {
 impl<P: Send + 'static> RcuReclaimer<P> {
     pub fn delay(mut self) {
         let ptr: P = unsafe {
-            let ptr = core::mem::replace(&mut self.ptr, core::mem::uninitialized());
+            // SAFETY. It is ok to have this uninitialized value because
+            // 1) Its memory won't be acccessed;
+            // 2) It will be forgotten rather than being dropped;
+            // 3) Before it gets forgtten, the code won't return prematurely or panic.
+            #[allow(clippy::mem_replace_with_uninit, clippy::uninit_assumed_init)]
+            let ptr = core::mem::replace(
+                &mut self.ptr,
+                core::mem::MaybeUninit::uninit().assume_init(),
+            );
 
             core::mem::forget(self);
 
             ptr
         };
-        get_singleton().after_grace_period(move || {
-            drop(ptr);
-        });
-    }
-}
 
-impl<P> Drop for RcuReclaimer<P> {
-    fn drop(&mut self) {
-        let wq = Arc::new(WaitQueue::new());
-        get_singleton().after_grace_period({
-            let wq = wq.clone();
-            move || {
-                wq.wake_one();
-            }
+        let rcu_monitor = RCU_MONITOR.get().unwrap().lock();
+        rcu_monitor.after_grace_period(move || {
+            drop(ptr);
         });
-        wq.wait_until(|| Some(0u8));
     }
 }
 
+/// Passes the quiescent state to the singleton RcuMonitor and take the callbacks if
+/// the current GP is complete.
+///
+/// # Safety
+///
+/// This function may disable the local IRQs.
```

Comment:
In Safety section, we explain the additional pre-condition to use an `unsafe` function safely.

> This function may disable the local IRQs.

This is NOT such a pre-condition. Here is my revised version.

```rust
/// Inform the RCU mechanism that the current task has reached a quiescent state.
/// A quiescent state is a point in the execution of a task at which 
/// it is guaranteed not to be holding any references to RCU-protected data.
///
/// # Safety
///
/// Determining whether the current task has reached a quiescent state
/// is fundamental for the RCU mechanism to work properly.
/// This responsibility falls on the shoulder of the caller of this function.
/// Failing to do so leads to undefined behaviors.
```

In addition, mark this function `pub(crate)` as it should never be used outside the `aster-frame`.